hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointStatisticsManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl;
+
+import java.util.Collection;
+import java.util.Map;
+
+public interface ClientEndpointStatisticsManager {
+
+    /**
+     * Called when the given endpoint is authenticated.
+     * @param endpoint the authenticated endpoint.
+     */
+    void onEndpointAuthenticated(ClientEndpoint endpoint);
+
+    /**
+     * Called when the given endpoint is destroyed. It
+     * might be the case that the endpoint is destroyed
+     * but not authenticated.
+     * @param endpoint the destroyed endpoint.
+     */
+    void onEndpointDestroyed(ClientEndpoint endpoint);
+
+    /**
+     * Returns the map of the statistics related to the
+     * clients connected to this node since the last time
+     * this method is called. Mapping is from the client
+     * type to statistics snapshot.
+     *
+     * @param activeEndpoints collection of clients that are
+     *                        active at that time.
+     * @return map of client type to statistics snapshots.
+     */
+    Map<String, ClientEndpointStatisticsSnapshot> getSnapshotsAndReset(Collection<ClientEndpoint> activeEndpoints);
```

Comment:
nit: this could be called `drainSnapshots()`

Reply:
If it is okay, I would like to keep this and the method name on the Counter class as they are